pkg/mount/mount.go

```diff
@@ -90,7 +126,7 @@ func ForceMount(device, target, mType, options string) error {
 // does a normal unmount.
 func Unmount(target string) error {
 	err := unmount(target, mntDetach)
-	if err == syscall.EINVAL {
+	if err != nil && errors.Cause(err) == syscall.EINVAL {
```

Comment:
Looks like we always want to ignore the `EINVAL` case (a similar check is in `RecursiveUnmount`); perhaps move this to `unmount()` and return `nil` if the error is due to `sys call.EINVAL`;

```patch
diff --git a/pkg/mount/mount.go b/pkg/mount/mount.go
index 12f968cbb8..3edd031acf 100644
--- a/pkg/mount/mount.go
+++ b/pkg/mount/mount.go
@@ -4,9 +4,7 @@ import (
 	"sort"
 	"strconv"
 	"strings"
-	"syscall"
 
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
@@ -125,12 +123,7 @@ func ForceMount(device, target, mType, options string) error {
 // Unmount lazily unmounts a filesystem on supported platforms, otherwise
 // does a normal unmount.
 func Unmount(target string) error {
-	err := unmount(target, mntDetach)
-	if err != nil && errors.Cause(err) == syscall.EINVAL {
-		// ignore "not mounted" error
-		err = nil
-	}
-	return err
+	return unmount(target, mntDetach)
 }
 
 // RecursiveUnmount unmounts the target and all mounts underneath, starting with
@@ -150,16 +143,6 @@ func RecursiveUnmount(target string) error {
 		logrus.Debugf("Trying to unmount %s", m.Mountpoint)
 		err = unmount(m.Mountpoint, mntDetach)
 		if err != nil {
-			// If the error is EINVAL either this whole package is wrong (invalid flags passed to unmount(2)) or this is
-			// not a mountpoint (which is ok in this case).
-			// Meanwhile calling `Mounted()` is very expensive.
-			//
-			// We've purposefully used `syscall.EINVAL` here instead of `unix.EINVAL` to avoid platform branching
-			// Since `EINVAL` is defined for both Windows and Linux in the `syscall` package (and other platforms),
-			//   this is nicer than defining a custom value that we can refer to in each platform file.
-			if errors.Cause(err) == syscall.EINVAL {
-				continue
-			}
 			if i == len(mounts)-1 {
 				if mounted, e := Mounted(m.Mountpoint); e != nil || mounted {
 					return err
diff --git a/pkg/mount/mounter_linux.go b/pkg/mount/mounter_linux.go
index bfa87a05ee..6f289e06d2 100644
--- a/pkg/mount/mounter_linux.go
+++ b/pkg/mount/mounter_linux.go
@@ -1,6 +1,9 @@
 package mount // import "github.com/docker/docker/pkg/mount"
 
 import (
+	"syscall"
+
+	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
@@ -77,6 +80,16 @@ func unmount(target string, flag int) error {
 	if err == nil {
 		return nil
 	}
+	// If the error is EINVAL either this whole package is wrong (invalid flags passed to unmount(2)) or this is
+	// not a mountpoint (which is ok in this case).
+	// Meanwhile calling `Mounted()` is very expensive.
+	//
+	// We've purposefully used `syscall.EINVAL` here instead of `unix.EINVAL` to avoid platform branching
+	// Since `EINVAL` is defined for both Windows and Linux in the `syscall` package (and other platforms),
+	//   this is nicer than defining a custom value that we can refer to in each platform file.
+	if errors.Cause(err) == syscall.EINVAL {
+		return nil
+	}
 	return &Error{
 		op:     "umount",
 		target: target,
```